### PR TITLE
Kriticism/add summary to openrec as submodule

### PR DESCRIPTION
**Pull request by kriticism**
# Description 

This is a big diff and does the following:
1. Adds a stripped down version of openrec (as a submodule) with a POC summary writer
2. Complete loop of the following steps:
     (a) On model creation, assign a random port for tensorboard visualization of this model.
     (b) Generate the models summary in a folder
     (c) Save this port and path of summary in the database against the model, so that tensorboard_reader can read.

# Test/Benchmark Outputs
TODO: remove the dataset that was added

# Checklist:
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Related Benchmarks have been performed, if applicable.
- [x] I have a plan to revert my changes.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if applicable.

### CODE REVIEW

File: board/models.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ class Model(EmbeddedDocument):
 	status = StringField(max_length=120, required=True)
 	dataset = EmbeddedDocumentField(Dataset)
 	logdir = StringField(max_length=120, required=True)	
+	port = StringField(max_length=8, required=True)
```
Comment (mohit-chawla) [inline board/models.py]: make this 5. #65536

File: board/views.py [hunk 2]
```diff
@@ -100,25 +100,57 @@ class HomePage(TemplateView):
     """
     template_name = 'home.html'
 
+LOG_DIR = "/Users/kay/recboard/model_summaries"
```
Comment (mohit-chawla) [inline board/views.py]: LOG_DIR = os.environ['RBROOT'] + "/model_summaries"

File: board/views.py [hunk 3]
```diff
@@ -100,25 +100,57 @@ class HomePage(TemplateView):
...
+    err = "No err "
+    user = get_dummy_user()
+    if not user.datasets: 
+        return JsonResponse({'msg':'No datasets found'})
+
+    if not user.workspaces:
+        user.workspaces.append(Workspace(name="workspacename2"))
+        db.insert('user',user)
+    
+    # TODO: maintain a list of available ports later
```
Comment (mohit-chawla) [inline board/views.py]: i started an issue for this. feel free to start issues for your TODO's. considering we are in hurry rn.

File: board/models.py (hunk 1 above)
Comment (kriticism) [reply board/models.py]: 👍

File: board/views.py (hunk 2 above)
Comment (kriticism) [reply board/views.py]: 👍

File: board/views.py (hunk 3 above)
Comment (kriticism) [reply board/views.py]: 👍

**Pull request merged**